tutorials/interactive/index.md

```diff
@@ -0,0 +1,814 @@
+
+Often it makes the most sense for the parent widget to manage the state
+and tell its child widget when to update. For example,
+[IconButton](https://docs.flutter.io/flutter/material/IconButton-class.html)
+allows you to treat an icon as a tappable button.
+IconButton is a stateless widget because the designers decided that
+the parent widget needs to know whether the button has been tapped,
+so it can take appropriate action.
+
+In the following example, TapboxB exports its state to its parent
+through a callback. Because TapboxB doesn't manage any state, it
+subclasses StatelessWidget.
+
+The ParentWidgetState class:
+
+* Manages the `_active` state for TapboxB.
+* Implements `handleTapboxChanged()`, the method called when the box is tapped.
+* Calls `setState` to update the UI.
+
+The TapboxB class:
+
+* Extends StatelessWidget because all state is handled by its parent.
+* When a tap is detected, it notifies the parent.
+
+<!-- _code/tapbox-b/main.dart -->
+<!-- skip -->
+{% prettify dart %}
+// ParentWidget manages the state for TapboxB.
+
+//------------------------ ParentWidget --------------------------------
+
+class ParentWidget extends StatefulWidget {
+  @override
+  _ParentWidgetState createState() => new _ParentWidgetState();
+}
+
+class _ParentWidgetState extends State<ParentWidget> {
+  bool _active = false;
+
+  void handleTapboxChanged(bool newValue) {
+    setState(() {
+      _active = newValue;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Container(
+      child: new TapboxB(
+        active: _active,
+        onChanged: handleTapboxChanged,
+      ),
+    );
+  }
+}
+
+//------------------------- TapboxB ----------------------------------
+
+class TapboxB extends StatelessWidget {
+  TapboxB({Key key, this.active: false, @required this.onChanged})
+      : super(key: key);
+
+  final bool active;
+  final ValueChanged<bool> onChanged;
+
+  void handleTap() {
+    onChanged(!active);
+  }
+
+  Widget build(BuildContext context) {
+    return new GestureDetector(
+      onTap: handleTap,
+      child: new Container(
+        child: new Center(
+          child: new Text(
+            active ? 'Active' : 'Inactive',
+            style: new TextStyle(fontSize: 32.0, color: Colors.white),
+          ),
+        ),
+        width: 200.0,
+        height: 200.0,
+        decoration: new BoxDecoration(
+          backgroundColor: active ? Colors.lightGreen[700] : Colors.grey[600],
+        ),
+      ),
+    );
+  }
+}
+{% endprettify %}
+
+**Dart code:**
+[`lib/main.dart`](https://raw.githubusercontent.com/flutter/website/master/_includes/_code/tapbox-b/main.dart)
+
+<aside class="alert alert-success" markdown="1">
+<i class="fa fa-lightbulb-o"> </i> **Tip:**
+When creating API, consider using the `@required` annotation for
+any parameters that your code relies on.
+To use `@required`, import the meta library:
+
+<pre>
+import package: meta/meta.dart;
```

Comment:
This is missing quotes, and has an extraneous space.

Also, I would recommend `'package:flutter/foundation.dart'` instead, which reexports meta.